components/EventItem.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useState, useEffect } from "react";
+import { StyleSheet, Text, View } from "react-native";
+import { Post } from "src/models";
+import IconButton from "./common/IconButton";
+import { ProfileImageAndName } from "./ProfileImageAndName";
+import { Divider } from "react-native-elements";
+
+interface Props {
+  item: Post,
+  writtenByYou: boolean,
+  startTime?: Date, 
+  maxOccupancy?: number,
+  hasInvitations: boolean,
+  eventColor: string
+}
+
+const EventItem = ({
+  item,
+  writtenByYou,
+  startTime,
+  maxOccupancy,
+  hasInvitations,
+  eventColor
+} : Props) => {
+  const [requested, setRequested] = useState(false); // If user has requested to join
+  const [numInvitations, setNumInvitations] = useState(0) // Number of requested invitations
+  const [isHours, setIsHours] = useState(true); // If time limit has >= 1 hour left
+  const [timeUntil, setTimeUntil] = useState(0); // Time (in either hours or minutes) until event starts
+  const [currentCapacity, setCurrentCapacity] = useState(1); // How many users have joined event
+  const [distance, setDistance] = useState(0); // Current distance user is from the event
+  const NUM_OF_LINES = 5;
+  const CAPACITY_PERCENTAGE = 0.75;
+
+  const setTime = () => {
+    if (startTime) {
+      const date = new Date()
+      const diffTime = Math.abs(startTime.getTime() - date.getTime());
+      const diffMin = Math.ceil(diffTime / (1000 * 60));
+      const diffHour = Math.ceil(diffTime / (1000 * 60 * 60));
+
+      if (diffMin < 60) {
+        setIsHours(false);
+        setTimeUntil(diffMin);
+
+      } else {
+        setIsHours(true);
+        setTimeUntil(diffHour);
+      }
+    } 
+  }
+
+  useEffect(() => {
+    setTime();
+  }, [startTime]);
+
+  const handleRequestToJoin = () => {
```

Comment:
I wonder if we should have a variation of the iconbutton that has a built-in toggleable state. But we can consider that later, for now this is very easy to read.